AGENT_NUMBER: 1
# 1 - Mode_1_Agent_8_FourBisection_DiagonalTarget 
# 3 - Mode_3_Agent_16_NoTarget
# 4 - Mode_4_Agent_1_RandomTarget
MAP_TYPE: 4

#collision threshold between robot and obstacle (or robot)
STATIC_COLLISION_TH: 0.2
DYNAMIC_COLLISION_TH: 0.45

# target threshold between target position and current position of robot
TARGET_TH: 0.3

# constraints for the robot
MAX_LINEAR_VAL: 1.0
MAX_ANGULAR_VAL: 1.0
MAP_MAX_RANGE: 6.0
MAP_MIN_RANGE: -6.0

# some flags
ENABLE_LASER_DETECTION: true
ENABLE_STATIC_COLLISIOM_TERMINAL: false
ENABLE_DYNAMIC_COLLISIOM_TERMINAL: true
ENABLE_EDGE_DETECTION: true
ENABLE_SURVIVE_REWARD: false
ENABLE_TARGET_TERMINAL: true

# reward related
TERMINAL_REWARD: 1.0             # arrive  
STATIC_COLLISION_REWARD:  -1.0   # static collision happen
DYNAMIC_COLLISION_REWARD: -20.0   # dynamic collision happen 
SURVIVE_REWARD: 1.0               # normal just discount
DISTANCE_REWARD: 0.0
EDGE_REWARD: 0.0

# reset position
ROBOT_X_START: -5.0
ROBOT_X_END: 5.0
ROBOT_Y_START: -5.0
ROBOT_Y_END: 5.0
ROBOT_YAW_START: -1.0
ROBOT_YAW_END: 1.0

# target position
TARGET_X_START: -5.0
TARGET_X_END: 5.0
TARGET_Y_START: -5.0
TARGET_Y_END: 5.0
